src/StlViewer/SceneSetup.tsx

```diff
@@ -128,49 +155,49 @@ const SceneSetup: React.FC<SceneSetupProps> = (
     const mesh = scene.getObjectByName('mesh') as Mesh
     const group = scene.getObjectByName('group') as Group
     const bbox = new Box3().setFromObject(mesh)
-    const height = bbox.max.z-bbox.min.z
-    group.position.z = height/2
+    const height = bbox.max.z - bbox.min.z
+    group.position.z = height / 2
   })
 
   const modelPosition: [number, number, number] = [
-    positionX ?? (meshDims.width*scale)/2,
-    positionY ?? (meshDims.length*scale)/2,
+    positionX ?? (meshDims.width * scale) / 2,
+    positionY ?? (meshDims.length * scale) / 2,
     0
   ]
 
   return (
-        <>
-            <scene background={BACKGROUND}/>
-            {sceneReady && showAxes && <axesHelper scale={[50, 50, 50]}/>}
-            {(cameraInitialPosition != null) && <Camera
-                initialPosition={cameraInitialPosition}
-                center={modelCenter}
-            />}
-            <Model3D
-                name={'group'}
-                meshProps={{ name: 'mesh' }}
-                scale={scale}
-                geometry={geometry}
-                position={modelPosition}
-                rotation={[rotationX, rotationY, rotationZ]}
-                visible={sceneReady}
-                materialProps={{ color }}
-                onLoaded={onLoaded}
-            />
-            <Floor
-                width={gridWidth ?? gridLength}
-                length={gridLength ?? gridWidth}
-                visible={sceneReady}
-                noShadow={!shadows}
-                offset={FLOOR_DISTANCE}
-            />
-            <Lights
-                distance={LIGHT_DISTANCE}
-                offsetX={modelPosition[0]}
-                offsetY={modelPosition[1]}
-            />
-            {sceneReady && orbitControls && <OrbitControls target={modelCenter} />}
-        </>
+    <>
+      <scene background={BACKGROUND} />
+      {sceneReady && showAxes && <axesHelper scale={[50, 50, 50]} />}
+      {(cameraInitialPosition != null) && <Camera
+        initialPosition={cameraInitialPosition}
+        center={modelCenter}
+      />}
+      <Model3D
+        name={'group'}
+        meshProps={{ name: 'mesh' }}
+        scale={scale}
+        geometry={subdivide > 0 ? subdividedGeometry : geometry}
```

Comment:
One way to avoid this additional logic could be to let the `useMemo` call above be:
```ts
() => subdivideProps !== undefined
      ? LoopSubdivision.modify(geometry, subdivideProps.subdivisions, subdivideProps)
      : geometry, // <- Return the unprocessed geometry
```
That way you could here just say:
```ts
geometry={subdividedGeometry}
```

If going this path, maybe changing the name `subdividedGeometry` to just `processedGeometry` for not making the assumption that the geometry is subdivided? 🤔